lib/wibox/layout/margin.lua

```diff
@@ -78,6 +78,14 @@ function margin:get_children()
     return {self.widget}
 end
 
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function margin:set_children(children)
+    self.widget = children and children[1]
```

Comment:
Right. Again, this is debatable and I have no killer argument. `check_widget` is useful in some "hot spot" in the code where all the magic may cause an apocalypse, but it is overkill to just use it everywhere.